appmonitoring/ts/src/CertificateManager.ts

```diff
@@ -436,10 +470,12 @@ export class CertificateManager {
          * and the webhook deployment is restarted. If neither certificate is regenerated, the reconciliation is complete.
          */
         if (shouldUpdate) {
-            await this.PatchWebhookAndSecretStore(operationId, kc, webhookCertData, clusterArmId, clusterArmRegion);
+            await this.UpdateWebhookAndSecretStore(operationId, kc, webhookCertData, clusterArmId, clusterArmRegion);
             if (shouldRestartDeployment) {
                 logger.info('Restarting webhook deployment so the pods pick up new certificates...', operationId, this.requestMetadata);
+                await logger.SendEvent("CertificateDeploymentRestartStarted", operationId, null, clusterArmId, clusterArmRegion, true);
                 await this.RestartWebhookDeployment(operationId, kc, clusterArmId, clusterArmRegion);
```

Comment:
What happens if this throws?

Reply:
then we must throw and fail. Otherwise the next run will think all is well